js/client-chat.js

```diff
@@ -1548,7 +1548,16 @@
 				this.$chat.append('<div class="message"><small>Loading...</small></div>');
 				this.$joinLeave = this.$chat.children().last();
 			}
-			this.joinLeave[action].push(user.group + user.name);
+
+			var formattedUser = user.group + user.name;
+			if (action === 'join' && this.joinLeave['leave'].includes(formattedUser)) {
+				this.joinLeave['leave'].splice(this.joinLeave['leave'].indexOf(formattedUser), 1);
+			} else if (action === 'leave' && this.joinLeave['join'].includes(formattedUser)) {
+				this.joinLeave['join'].splice(this.joinLeave['join'].indexOf(formattedUser), 1);
```

Comment:
```suggestion
```

In case my outlining wasn't clear:

- If a user leaves and rejoins, this can be removed completely
- If a user joins and leaves, this should be kept, because it's useful to know that someone was in the room briefly

And the message ordering always puts leaves after joins, so you don't have to worry about the order beyond that.

Reply:
Oh, I misunderstood that.